q2_moshpit/prodigal/prodigal.py

```diff
@@ -24,34 +26,56 @@ def predict_genes_prodigal(
     genes = GenesDirectoryFormat()
     proteins = ProteinsDirectoryFormat()
 
-    # Get paths to fasta files in input dir
-    fasta_files = [
-        file for file in os.listdir(mags.path)
-        if file.endswith(".fa") or file.endswith(".fasta")
-    ]
-
     # Define base command
     base_cmd = [
         "prodigal",
         "-g", translation_table_number,
         "-f", "gff"
     ]
 
-    # For every fasta file in mags.path call prodigal and write
-    # outputs to the corresponding directories.
-    for fasta_file in fasta_files:
-        # Get the filename from the file path
-        file_id = os.path.splitext(fasta_file)[0]
-
-        # Adjust command and run
-        cmd = cp.deepcopy(base_cmd)
-        cmd.extend([
-            "-i", os.path.join(mags.path, fasta_file),
-            "-o", os.path.join(loci.path, f"{file_id}_loci.gff"),
-            "-a", os.path.join(proteins.path, f"{file_id}_proteins.fasta"),
-            "-d", os.path.join(genes.path, f"{file_id}_genes.fasta")
-        ])
-        run_command(cmd)
+    def _process_fasta_files(fasta_files: list, prefix: str, input_path: str):
+        # For every fasta file call prodigal and write
+        # outputs to the corresponding directories.
+        for fasta_file in fasta_files:
+            # Get the filename from the file path
+            file_id = os.path.splitext(fasta_file)[0]
+
+            # Adjust command and run
+            cmd = cp.deepcopy(base_cmd)
+            cmd.extend([
+                "-i", os.path.join(input_path, fasta_file),
+                "-o",
+                os.path.join(
+                    loci.path, f"{prefix}{file_id}_loci.gff"
+                ),
+                "-a",
+                os.path.join(
+                    proteins.path, f"{prefix}{file_id}_proteins.fasta"
+                ),
+                "-d",
+                os.path.join(
+                    genes.path, f"{prefix}{file_id}_genes.fasta"
+                )
+            ])
+            run_command(cmd)
```

Comment:
I think this function should be rewritten slightly to separate the filename processing logic from running the actual prodigal command. Instead of accepting a list of fasta file paths, it should take a single path, a file "id" and the prefix and construct the desired command.